### ISSUE TITLE
Training loss curve osciliates

### ISSUE THREAD

**Issue by Happylkx** [tloen/alpaca-lora#246]
I am finetuning an alpaca-7B model , but the training loss seems to osciliate in a fixed period. Is this normal?

![image](https://user-images.githubusercontent.com/10860589/229297950-e704024f-f4b6-4114-ba89-d29f3ddfe22f.png)

Here is the command I used to train the model:
```bash
python finetune_code.py \
    --base_model='decapoda-research/llama-7b-hf' \
    --num_epochs=10 \
    --cutoff_len=1024 \
    --group_by_length \
    --output_dir='./lora-alpaca-code' \
    --lora_target_modules='[q_proj,k_proj,v_proj,o_proj]' \
    --lora_r=16 \
    --micro_batch_size=8```

**Comment by AngainorDev** [comment]
The recurring pattern is due to "--group_by_length" that groups the samples by len instead of a random order.
I do advise to leave that switch off.

**Comment by Happylkx** [comment]
> The recurring pattern is due to "--group_by_length" that groups the samples by len instead of a random order. I do advise to leave that switch off.

Solved. Thanks!

**Comment by Happylkx** [comment]
> > > The recurring pattern is due to "--group_by_length" that groups the samples by len instead of a random order. I do advise to leave that switch off.
> > 
> > 
> > Solved. Thanks!
> 
> Hi @Happylkx What type of GPU do you use to fine-tune LLama, V100-32G, V100-16, A100 or 3090?

I used an A10 card with 24 GB VRAM. I think 16GB should also work, but you probably need deepspeed.

**Comment by OpenSource-fan** [comment]
Thanks for your response. I use the 4*V100*32G and remove `--group_by_length`. The training loss often becomes suddenly bigger and then it goes to 0. I have tried more than 10 times.
![image](https://user-images.githubusercontent.com/81273096/231821076-c3b9ca78-4805-44a4-8daf-71aef71391a7.png)
